.github/workflows/build.yml

```diff
@@ -0,0 +1,31 @@
+name: "Build"
+on: [push, pull_request]
+jobs:
+  build:
+    runs-on: ${{matrix.os}}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+    steps:
+      - uses: actions/checkout@v2.4.0
+        with:
+          # Nix Flakes doesn't work on shallow clones
+          fetch-depth: 0
+      - uses: cachix/install-nix-action@v15
+        with:
+          install_url: https://releases.nixos.org/nix/nix-2.5.1/install
+          extra_nix_config: |
+            experimental-features = nix-command flakes
+      - name: nix build
+        run: |
+          nix build
+      - name: nix log
+        if: ${{ always() }}
+        run: |
+          nix log &> ${{ matrix.os }}-souffle-haskell.log
+      - name: upload ${{ matrix.os }}-souffle-haskell.log
+        if: ${{ always() }}
```

Comment:
Line can be removed?

Reply:
Let's keep this, otherwise you'd have no log to check whenever the build above fails

Reply:
No, I didn't mean the upload itself, I meant the if condition.  "if always/true" seems redundant?

Reply:
By default, actions are run if the previous ones succeeded. `if ${{always()}}` here is required to change this behavior; we do need this I'm afraid.

Reply:
Ok, didn't know that!